dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/Input.java

```diff
@@ -0,0 +1,67 @@
+package com.dotcms.rest.api.v1.apps;
+
+import com.dotcms.repackage.javax.validation.constraints.NotNull;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * This class represent an input value captured using a char array The value on the char array can
+ * be wiped out as opposed to a string which are final
+ */
+@JsonDeserialize(using = Input.InputDeserialize.class)
+public class Input {
+
+    @NotNull
+    private final char[] value;
+
+    private final boolean hidden;
+
+    private Input(final char[] value, final boolean hidden) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The user-supplied array 'value' is stored directly.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5289044)